[Reagan Tan Wei Liang] iP

ReaganTan00 opened the pull request:
### DukeRT ###

> "It will probably get worse" - Probably not Confucius 

DukeRT is a simple chat bot that tracks your tasks. It's, 

- text-based
- easy to learn
- ~~painful~~ fun to use!

All you need to do is,

1. download it from [here](https://github.com/ReaganTan00/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 🔥😮 

And it is **FREE!**

Features: 

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Below is the Duke.java file implementation. Here's the `main` method:
```rust
public class Duke {
    public static void main(String[] args) throws DukeException {
        Ui.greet();
        Scanner sc = new Scanner(System.in);
        Storage storage = new Storage();
        try {
            storage.storageRead();
        } catch (IOException e) {
            e.printStackTrace();
        }
        TaskList taskList = storage.getTaskList();
        while (true) {
            try {
                String input = sc.nextLine();
                Command command = Parser.parseCommand(input);
                command.run(taskList);
            } catch (DukeException e) {
                Ui.formatMessage(e.toString());
            }
        }
    }
}
```

Commit:
Add Gradle support

ReaganTan00 committed:
Level-1

ReaganTan00 committed:
Level-2

ReaganTan00 committed:
Level-3

ReaganTan00 committed:
Level-4

ReaganTan00 committed:
A-TextUiTesting

ReaganTan00 committed:
Level-5

ReaganTan00 committed:
Level-6

ReaganTan00 committed:
Level 7

ReaganTan00 committed:
Missing duke.txt??

ReaganTan00 committed:
Clean up of code

ReaganTan00 committed:
Level 8

ReaganTan00 committed:
Merge branch 'master' into branch-Level-7

ReaganTan00 committed:
Attempt to clean up merge

ReaganTan00 committed:
Merge branch 'branch-Level-7'

ReaganTan00 committed:
Resolve merge conflicts

ReaganTan00 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

ReaganTan00 committed:
A-Gradle

ReaganTan00 committed:
A-MoreOOP

ReaganTan00 committed:
A-Packages

ReaganTan00 committed:
A-JUnit

ReaganTan00 committed:
A-Javadoc

ReaganTan00 committed:
Forgot to branch while working on code violations

ReaganTan00 committed:
A-CodingStandard

ReaganTan00 committed:
Level-9

ReaganTan00 committed:
Merge branch 'branch-Level-9'

ReaganTan00 committed:
Merge branch 'branch-A-CodingStandard'

ReaganTan00 committed:
Resolved merge conflicts
Merge branch 'branch-A-Javadoc'

# Conflicts:
#	src/main/java/duke/task/Task.java

ReaganTan00 committed:
A-Checkstyle

ReaganTan00 committed:
Merge branch 'branch-A-CheckStyle'

stevenlimhw commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,131 @@
+     *
+     * @param taskList the task list
+     */
+    public void storageWrite(ArrayList<Task> taskList) {
```
The name of the method is better written as `writeStorage`, to clearly indicate action.

stevenlimhw commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,131 @@
+     *
+     * @throws IOException the io exception
+     */
+    public void storageRead() throws IOException {
```
The method name should be written as `readStorage`.

The code base seems to adhere to the coding standard for the most part. Some minor mistakes include the naming of the methods.

Moreover, I think that it is unnecessary to push the `.class` files onto the GitHub repository. Perhaps, you can add `.class` files into your `.gitignore` file.

ReaganTan00 replied:
Made the changes accordingly, thank you! 👍

stevenlimhw reviewed (commented):
The code base seems to adhere to the coding standard for the most part. Some minor mistakes include the naming of the methods.

Moreover, I think that it is unnecessary to push the `.class` files onto the GitHub repository. Perhaps, you can add `.class` files into your `.gitignore` file.

ReaganTan00 committed:
Level-10
Currently still buggy

ReaganTan00 committed:
Merge branch 'branch-Level-10'

ReaganTan00 committed:
Cleaned up Level 10

ReaganTan00 committed:
Coding standard changes, and added *.class files to .gitignore

ReaganTan00 committed:
Removed *.class files

paotheroo commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+    /**
+     * Gets task from a given index.
+     *
+     * @param i the
```
What is i?

ReaganTan00 committed:
Updated DialogBox

ReaganTan00 committed:
Added assertions for the following commands:
*deadline
*delete
*event
*find
*mark
*unmark

Storage class: added assertions

Ensure reading and writing of tasks done in appropriate format

Parser class: added assertions

Commands that do not require extra input etc. bye, list will now be
checked through assertions

Minor code clean up

ReaganTan00 committed:
Command classes that violated code quality and were refactored:
*Deadline
*Event
*Find
*Todo

Code quality was violated, following refactorings were done:
*Decompose Conditional
*Replace Magic Literal
*Reverse Conditional

Added segments of input as a class field for relevent commands.

ReaganTan00 committed:
Merge pull request #1 from ReaganTan00/branch-A-Assertions

Added assertions for the following commands:

ReaganTan00 committed:
Merge branch 'master' of https://github.com/ReaganTan00/ip

* 'master' of https://github.com/ReaganTan00/ip:
  Added assertions for the following commands: *deadline *delete *event *find *mark *unmark

ReaganTan00 committed:
Merge branch 'master' into heads/branch-A-CodeQuality

ReaganTan00 committed:
Merge pull request #2 from ReaganTan00/heads/branch-A-CodeQuality

Command classes that violated code quality and were refactored:

ReaganTan00 committed:
Merge branch 'master' of https://github.com/ReaganTan00/ip

* 'master' of https://github.com/ReaganTan00/ip:
  Command classes that violated code quality and were refactored: *Deadline *Event *Find *Todo

ReaganTan00 committed:
Cleaned up merge formatting to pass checkstyle

ReaganTan00 committed:
TagCommand class: Added a new command to tag tasks

To increase functionality of the chat bot Duke by allowing tasks
to be tagged according to what the user likes.

Other classes augmented to help store the new functionality:
*Deadline
*Event
*Task
*Todo
*Parser
*Storage

C-Tagging

ReaganTan00 committed:
Add User Guide and representative screenshot.

As per Week 6 iP increment, edited root/docs/README.md and included
a representative screenshot as root/docs/Ui.png.

Added expected output images to root/docs/ for the User Guide.

Small code quality changes made to:
*Parser.java
*TaskList.java
*Ui.java

This is to align method headers/comments to SE-Guidelines.

Formatting change made to FindCommand to tidy up output formats.

Minute changes made to *.FXML files to improve user interface.

ReaganTan00 committed:
Update README.md

ReaganTan00 committed:
Update README.md

ReaganTan00 committed:
Update README.md

ReaganTan00 committed:
Unable to produce the JAR file appropriately.

Main no longer extends Application, used a wrapper class labeled as
'MainWrapper'.

ReaganTan00 committed:
Merge branch 'master' of https://github.com/ReaganTan00/ip

* 'master' of https://github.com/ReaganTan00/ip:
  Update README.md
  Update README.md
  Update README.md

ReaganTan00 committed:
Minor java documentation changes to pass checkstyle.
